pyhf/pdf.py

```diff
@@ -108,7 +114,7 @@ def param_set(self, name):
         return self.par_map[name]['paramset']
 
     def set_poi(self, name):
-        if name not in [x for x, _ in self.modifiers]:
+        if name not in [x for x, _, _ in self.modifiers]:
```

Comment:
this line is incorrect. we want `if name not in [x for _, _, x in self.modifiers]` and really, it should just become `if not not in self.parameters`.